src/ramptr.c

```diff
@@ -22,7 +22,46 @@ void sub_08012CDC(void)
     CpuFill16(0, gUnk_02000540, 0x1E4);
 }
 
-void sub_08012D14(u16 r0, u32 *r1)
+void sub_08012D14(u16 r0, u16 *r1)
 {
     gUnk_03006590[r0].unk28 = r1;
 }
+
+void sub_08012D30(u32 r0)
+{
+    LZ77UnCompWram(sub_08001D3C(2,r0), gUnk_03006590[0].unk28);
+    sub_08012F0C(4, r0);
+    sub_08012F0C(0x180, 0);
+}
+
+void sub_08012D64(u16 idx, u16 r1, u32 r2)
+{
+    gUnk_03006590[idx].unk0 |= (1 << r1);
+    gUnk_03006590[idx].unk2C[r1].unk0 = 1;
+    gUnk_03006590[idx].unk2C[r1].unk4 = (u16*)((u32)gUnk_03006590[idx].unk28 + 16 + (r2 & 0xfffe)); // Looks weird. The sequence of operators affects code generation. 
```

Comment:
This is by no means readable. unk28 is obviously a pointer so why not type it properly?